### DISCUSSION TITLE
Need helping with parsing error

### DISCUSSION THREAD

**Discussion by giveitawhirl** [ReagentX/imessage-exporter#180] [Q&A] [answered]
No functions work as I receive the following when initiating a process

`Unable to launch: Failed to parse row: Invalid column type Null at index: 7, name: service_name`

I was unable to find something in the documentation, any help would be greatly appreciated.

Thanks!

**Comment by ReagentX** [comment, marked as answer]
How old is this database? Was it created pre-iMessage?

This is caused by `imessage-exporter` not parsing the `service_name` as `Option<String>`:

https://github.com/ReagentX/imessage-exporter/blob/4ccea214e4e775f23a4d1da97f1d5d9319fc8505/imessage-database/src/tables/chat.rs#L19

**Reply by ReagentX** [reply]
I am tracking this bug in #181.

**Reply by giveitawhirl** [reply]
I don't believe so, how would I check?

**Reply by ReagentX** [reply]
This will be fixed in the next release, 1.7 Laurel Sumac. 

Edit: Laurel Sumac is [now available](https://github.com/ReagentX/imessage-exporter/releases/tag/1.7.0)!

**Comment by rlxmp** [comment]
Looking forward for this release for this fix too, thank you!

**Reply by ReagentX** [reply]
Laurel Sumac is now available: https://github.com/ReagentX/imessage-exporter/releases/tag/1.7.0
